### PR TITLE
added pooled miner result polling support

### PR DESCRIPTION
**Pull request by soualid**
This PR adds a feature to poll `bxxd.eth` pooled miner results and submit a mint transaction if the pool finds a nonce for the current wallet public address.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -199,6 +208,46 @@ const LICENSE_ENV_VARIABLES = [
...
+      }
+
+    }
+    await sleep(5000);
+  }
+}
+
+if (process.env.POLL_POOLED_MINER_RESULTS == 'true') {
+  pollPooledMinerResults();
+}
```
Comment (coltmeister) [inline src/index.ts:249]: Let's put this within the `app.listen` `try { } catch { }` block!

File: src/index.ts [hunk 2]
```diff
@@ -199,6 +208,46 @@ const LICENSE_ENV_VARIABLES = [
...
+  while (true) {
+    let result = await axios.get<PooledMinerResult>(process.env.POOLED_MINER_RESULT_URL);
+    let last = result.data;
+    console.log(`polling pooled miner results for address ${wallet.address}, last mined address is ${last.address}...`)
+    if (last.address === wallet.address &&
+      last.error === null &&
+      last.success === true &&
+      submittedNonces.indexOf(last.nonce) == -1) {
+      console.log('found a valid nonce, will mint it if gas price is low enough');
+      const nonce = BigNumber.from(last.nonce);
```
Comment (coltmeister) [inline src/index.ts:224]: Just to be safe, I think we should call `checkNonce` here as well and confirm that the returned `nonceStatus` is `VALID` (similar to the logic in the `submit-work` endpoint)

**Review by coltondotio — CHANGES_REQUESTED** [review]
This is pretty cool -- thanks for adding this functionality for the pool mining folks!

File: src/index.ts (hunk 1 above)
Comment (soualid) [reply src/index.ts:249]: Yup, way better.

File: src/index.ts (hunk 2 above)
Comment (soualid) [reply src/index.ts:224]: Donee

**Review by coltondotio — APPROVED** [review]

**Pull request merged**